### PR TITLE
MGMT-17876: Move device.spec.templateVersion to an annotation

### PR DESCRIPTION
**Pull request by avishayt**

### CODE REVIEW

File: internal/tasks/fleet_validate.go [hunk 1]
```diff
@@ -116,7 +116,7 @@ func (t *FleetValidateLogic) validateFleetTemplate(ctx context.Context, fleet *a
 	return retErr
 }
 
-func (t *FleetValidateLogic) validateConfigItem(ctx context.Context, configItem *api.DeviceSpecification_Config_Item) (string, error) {
+func (t *FleetValidateLogic) validateConfigItem(ctx context.Context, configItem *api.DeviceSpec_Config_Item) (string, error) {
```
Comment (hexfusion) [inline internal/tasks/fleet_validate.go:119]: does openAPi generate these names just curious why this is not api.DeviceSpecConfigItem. Is there a config to change that?

File: internal/tasks/fleet_rollout.go [hunk 2]
```diff
@@ -158,14 +158,24 @@ func (f FleetRolloutsLogic) RolloutDevice(ctx context.Context) error {
...
-	if device.Spec.TemplateVersion != nil && *device.Spec.TemplateVersion == *templateVersion.Metadata.Name {
+	currentVersion := ""
+	if device.Metadata.Annotations != nil {
+		v, ok := (*device.Metadata.Annotations)[model.DeviceAnnotationTemplateVersion]
+		if ok {
+			currentVersion = v
+		}
+	}
+
+	if currentVersion == *templateVersion.Metadata.Name {
```
[the remark this answers is not in the document]
Comment (avishayt) [reply internal/tasks/fleet_rollout.go:169]: If currentVersion is empty (which would always happen for a new device), then we will update the device.

File: internal/tasks/fleet_validate.go (hunk 1 above)
Comment (avishayt) [reply internal/tasks/fleet_validate.go:119]: Yes, the code generator generates that on its own.

File: internal/tasks/fleet_rollout.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (hexfusion) [reply internal/tasks/fleet_rollout.go:169]: It makes sense should we add a comment? If its empty and its not a new device should we log a warning?

**Review by mangelajo — APPROVED** [review]
It looks good, thanks for handling this.

**Pull request merged**